testing/test_service_assertions.py

```diff
@@ -0,0 +1,231 @@
+import os
+from abc import ABC, abstractmethod
+
+import numpy as np
+import pytest
+from sklearn.metrics import mean_squared_error
+
+from neuraxle.base import Identity, ExecutionContext, ForceHandleMixin, StepWithContext
+from neuraxle.data_container import DataContainer
+from neuraxle.metaopt.auto_ml import InMemoryHyperparamsRepository, AutoML, RandomSearchHyperparameterSelectionStrategy, \
+    ValidationSplitter
+from neuraxle.metaopt.callbacks import ScoringCallback
+from neuraxle.pipeline import Pipeline
+
+
+class BaseService(ABC):
+    @abstractmethod
+    def service_method(self, data):
+        pass
+
+
+class SomeService(BaseService):
+    def service_method(self, data):
+        self.data = data
+
+
+# class SomeStepThatChangesTheRootOfTheExecutionContext(ForceHandleMixin, Identity):
+#     def __init__(self):
+#         Identity.__init__(self)
+#         ForceHandleMixin.__init__(self)
+#
+#     def _will_process(self, data_container: DataContainer, context: ExecutionContext):
+#         context.root = 'invalid_root'
+#         super()._will_process(data_container, context)
+#         return data_container, context
+#
+#     def _transform_data_container(self, data_container: DataContainer, context: ExecutionContext):
+#         return data_container
+
+
+class SomeStep(ForceHandleMixin, Identity):
```

Comment:
Such a force handle identity seems to be used in other test files as well. Creating a `ForceHandleIdentity` next to the real `Identity` in the framework may be good to import it in tests.

Reply:
It would sure be useful. Although now that I look at it, there is also the IdentityHandlerMethodsMixin which seems to serve a similar purpose.

Reply:
Done. I did the modifications it in the wrong branch though so it will be part of the next PR instead of the revision of this one.